Tasks/NuGetInstaller/nugetinstaller.ts

```diff
@@ -43,61 +46,50 @@ async function main(): Promise<void> {
         });
 
         let noCache = tl.getBoolInput("noCache");
-        let nuGetRestoreArgs = tl.getInput("nuGetRestoreArgs");
         let verbosity = tl.getInput("verbosity");
 
-        let restoreMode = tl.getInput("restoreMode") || "Restore";
-        // normalize the restore mode for display purposes, and ensure it's a known one
-        let normalizedRestoreMode = ["restore", "install"].find(x => restoreMode.toUpperCase() === x.toUpperCase());
-        if (!normalizedRestoreMode) {
-            throw new Error(tl.loc("UnknownRestoreMode", restoreMode));
-        }
-
-        restoreMode = normalizedRestoreMode;
+        let packagesDirectory = tl.getPathInput("packagesDirectory");
 
-        let nugetConfigPath = tl.getPathInput("nugetConfigPath", false, true);
-        if (!tl.filePathSupplied("nugetConfigPath")) {
-            nugetConfigPath = null;
+        if (!tl.filePathSupplied("packagesDirectory")) {
+            packagesDirectory = null;
         }
 
-        let nugetUxOption = tl.getInput('nuGetVersion');
-
-        // due to a bug where we accidentally allowed nuGetPath to be surrounded by quotes before,
-        // locateNuGetExe() will strip them and check for existence there.
-        let nuGetPath = tl.getPathInput("nuGetPath", false, false);
-        let userNuGetProvided = false;
-        if(nuGetPath !== null && tl.filePathSupplied("nuGetPath")){
-            nuGetPath = nutil.stripLeadingAndTrailingQuotes(nuGetPath);
-            // True if the user provided their own version of NuGet
-            userNuGetProvided = true;
-            if (nugetUxOption !== "custom"){
-                // For back compat, if a path has already been specificed then use it.
-                // However warn the user in the build of this behavior
-                tl.warning(tl.loc("Warning_ConflictingNuGetPreference"));
+        let source : string = null;
+        let nuGetConfigPath : string = null;
+        let selectOrConfig = tl.getInput("selectOrConfig");
+        if (selectOrConfig === "select" ) {
+            let feeds : Array<string> = [];
+            let feed = tl.getInput("feed");
+            if (feed) {
+                feeds.push(feed);
```

Comment:
I think we might be missing a step here. If I understand correctly feed = {name: string; id: string;}. We would need to get the NuGet registry URL with that info.